src/Symfony/Component/Clock/HrClock.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * A high-resolution monotonic clock, counting from an arbitrary point in time.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class HrClock implements TimeInterface
+{
+    private \DateTimeZone $timezone;
+
+    public function __construct(\DateTimeZone|string $timezone = null)
+    {
+        if (false === hrtime()) {
+            throw new \LogicException('hrtime() returned false: the runtime environment does not provide access to high-resolution time.');
+        }
+
+        if (\is_string($timezone ??= date_default_timezone_get())) {
+            $this->timezone = new \DateTimeZone($timezone);
+        } else {
+            $this->timezone = $timezone;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function now(): \DateTimeImmutable
+    {
+        $hrtime = hrtime();
+        $now = '@'.$hrtime[0].'.'.substr(str_pad($hrtime[1], 9, 0, \STR_PAD_LEFT), 0, -3);
+
+        return (new \DateTimeImmutable($now, $this->timezone))->setTimezone($this->timezone);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function timeArray(): array
+    {
+        $now = hrtime();
+
+        return [$now[0], (float) ($now[1] / 1E3)];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function timeFloat(): float
+    {
+        return hrtime(true) / 1E9;
```

Comment:
What about using named arguments `hrtime(as_number: true)`?

Reply:
I don't think that's needed. The doc is here if ppl wonder about the arguments of a function.